src/command/restore/restore.c

```diff
@@ -2073,6 +2074,32 @@ restoreProcessQueue(const Manifest *const manifest, List **const queueList)
             const ManifestFilePack *const filePack = manifestFilePackGet(manifest, fileIdx);
             const ManifestFile file = manifestFileUnpack(manifest, filePack);
 
+            Oid dbNode, relNode;
+            if (sscanf(strZ(file.name), MANIFEST_TARGET_PGDATA "/" PG_PATH_BASE "/%u/%u", &dbNode, &relNode) == 2)
+            {
+                if (!isRelationNeeded(dbNode, DEFAULTTABLESPACE_OID, relNode))
+                    continue;
+            }
+            else if (strBeginsWithZ(file.name, MANIFEST_TARGET_PGTBLSPC "/"))
+            {
+                StringList *lst = strLstNewSplitZ(file.name, "/");
+                if (strLstSize(lst) == 5)
+                {
+                    Oid spcNode;
+                    if (
+                        // Check all sscanf-s at once to simplify the tests
+                        sscanf(strZ(strLstGet(lst, 1)), "%u", &spcNode) +
```

Comment:
Can this line return 0? If so, why do we need to check the other lines?

Reply:
Yes, it can. The current solution is used to simplify the tests.

Reply:
But are tests the first priority?